processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -127,45 +160,44 @@ protected Integer addToFacts(
       ThreadLocal<InputRow> in
   ) throws IndexSizeExceededException
   {
-    Integer rowOffset;
-    synchronized (this) {
-      rowOffset = numEntries.get();
-      if(rowOffset >= maxRowCount && !facts.containsKey(key)) {
-        throw new IndexSizeExceededException("Maximum number of rows reached");
-      }
-      final Integer prev = facts.putIfAbsent(key, rowOffset);
-      if (prev != null) {
-        rowOffset = prev;
-      } else {
-        Aggregator[] aggs = new Aggregator[metrics.length];
-        for (int i = 0; i < metrics.length; i++) {
-          final AggregatorFactory agg = metrics[i];
-          aggs[i] = agg.factorize(
-              makeColumnSelectorFactory(agg, in, deserializeComplexMetrics)
-          );
-        }
-        aggList.add(aggs);
-        numEntries.incrementAndGet();
+    final Aggregator[] aggs;
+    final Integer retval;
+    final Integer rowOffset = numEntries.get();
```

Comment:
in the 0.6.x branch there's two sections of synchronization in the add row method, one which is still in IncrementalIndex, and then this one:

```
    synchronized (this) {
      in = row;
      for (Aggregator agg : aggs) {
        agg.aggregate();
      }
      in = null;
    }
```

The later seems to only be synchronizing on `in`, which is a ThreadLocal variable now. 

I don't see any other items synchronizing on `IncrementalIndex` So I'm pretty sure this section of code was only synchronized for local purposes.